e2e/framework.go

```diff
@@ -188,11 +188,25 @@ func (c *cluster) WaitForHeight(num uint64, timeout time.Duration, nodes ...[]st
 				return
 			}
 		case <-timer.C:
-			c.t.Fatal("timeout")
+			if throwPanic {
```

Comment:
Why do you want to throwPanic here? What is the workflow of the tests that use this panic?

Reply:
This is added because we need somehow to detect timeout in tests. We found that it is easier to detect timeout in tests with panic/recover and we thought `WaitForHeight` is just a simple function inside e2e testing framework so introducing one more parameter is good enough solution. Otherwise the only other solution is to mock logs and catch t.Fatal("timeout")